### ISSUE TITLE
Trade Error between ShinPokeRed / Perfect Crystal rom hacks

### ISSUE THREAD

**Issue by DiabloStorm** [jojobear13/shinpokered#268]
So, again this may not be something you can do much about, but here's another trade issue since the last time we talked about this, manifesting in a different way this time. The trade completed, but with data loss (no I didn't screw up bgb this time like with yellow) Instead of the "An abnormality was detected, trade cancelled" (paraphrasing)

Using my old save from a previous patch (v1.23.11m) with ShinRed trading shinred horsea with perfect crystal (current github version) pikachu

Shin pov:
![image](https://user-images.githubusercontent.com/63818362/226494268-a4a05b65-4746-4e37-b4f1-90c2349f109b.png)

Perf crystal pov: Name, stats and ID corruption

![bgb00168](https://user-images.githubusercontent.com/63818362/226495147-a4ca5080-f96a-4a1f-955c-131fff04962f.gif)
![bgb00169](https://user-images.githubusercontent.com/63818362/226495148-cc3f0074-6dda-41bb-9d23-2f161a98714d.gif)

Horsea also corrupted:

![bgb00170](https://user-images.githubusercontent.com/63818362/226495345-2be7569c-b6b5-4114-b6fc-9be458bf3126.gif)

Save states before trade:
[Save States.zip](https://github.com/jojobear13/shinpokered/files/11024102/Save.States.zip)

Out of curiosity I tried trading the mons I had issues with originally, no "abnormality" error message, but they transfer with data corruption now.

**Comment by jojobear13** [comment]
Revisited this issue, starting again fresh, after being stumped for a while. I think I discovered the root of the problem.

According to Perfect Crystal's github page:
https://github.com/Superegz/Pokemon-Perfect-Crystal/tree/master#the-attached-ips-works-with-pokemon---crystal-version-ue-v11-c-and-the-patched-rom-should-have-a-checksum-of-c448
Perfect crystal needs to be patched onto a Crystal Version USA/EU V1.1 rom. This vanilla rom has a CRC checksum of 3358e30a.

When I took the attached perfect crystal save state and loaded up the perfect crystal rom, BGB said that the hash of the save state's rom did not match the loaded rom. 

Out of curiosity, I re-patched Perfect Crystal onto a Crystal Version USA V1.0 rom having CRC checksum ee6f5188. The checksum this time DID match with the attached save state.

TLDR, Perfect Crystal was patched using the wrong version of the base crystal rom. This can cause unforeseen errors, and particularly so with the trade center considering the changes between V1.0 and V1.1 of crystal.

**Comment by DiabloStorm** [comment]
You're a heck of a detective. I didn't realize I could even patch it incorrectly.
